### ISSUE TITLE
Task Changes - prerequiste tasks updates - updates to comment.

### ISSUE THREAD

**Issue by fergalddaly** [Keroberos/sam#76]
### VAT Return Task Changes

Change so that VAT return is the first task to be completed.
Set a due date on the VAT return - 17th of Month of VAT Return
All the task from previous months have to be completed before a vat return task can be changed to TODO.

The very first VAT return has to be manually entered (as the data in the system simply is not there to do it automatically).  this will have to be handled on the VAT Return dashboard.  -- create a change for this.

When the status of the monthly tasks is updated to completed, then set the VAT return task to TODO (as it can now be completed).

## Invoice Tasks
Must issue invoices in order

## Bank Upload Tasks
Must upload in order

## Bank Reconciliation Task Changes
Change to TODO if 
the previous 2 months are complete
the bank transactions have been upload.

**Comment by fergalddaly** [comment]
@Keroberos - I have made quite a few changes to the tasks so that tasks are now completed in the proper order - that the prerequisite tasks are completed first.  The comment section now reflects what the user must do.

While the above all works fine, I do think that there has been quite a bit of "manual" connecting tasks.
While this is ok for now and it does all work, the tasks section really needs to redone.
A more generic approach of task mgt would be much better.
So basically, when the tasks are created initially, then the prerequiste tasks are created too, represented as objects rather than "hardcoded" in the code.  
I think we can leave this for a bit and move onto more pressing needs for now.

**Comment by fergalddaly** [comment]
Commits:

e04e95fc661e33a8491f35a0ccd9bb031b7e6833

5ac2bcdc8e56cb6961f92cd1f106af9e591c314d

325dd9e1ee7dba4f07540e0e7bee4468dc8d5e8d  (note: forgot to update comment on this commit).

**Comment by fergalddaly** [comment]
merged to master branch.
